ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/siteConfig/siteConfigBuilder.go

```diff
@@ -69,6 +76,17 @@ func (scbuilder *SiteConfigBuilder) getClusterCRs(clusterId int, siteConfigTemp
 	for _, cr := range scbuilder.SourceClusterCRs {
 		mapSourceCR := cr.(map[string]interface{})
 
+		if mapSourceCR["kind"] == "AgentClusterInstall" {
+			networkType := siteConfigTemp.Spec.Clusters[clusterId].NetworkType
+			crMetadata := mapSourceCR["metadata"].(map[string]interface{})
+			networkTypeOverride := "{\"networking\":{\"networkType\":\"" + networkType + "\"}}"
+			annotations := make(map[string]interface{})
+			// This overrides existing annotations, at this point the ACI definition
+			// in the cluster-crs.yaml file does not have any
+			annotations["agent-install.openshift.io/install-config-overrides"] = networkTypeOverride
```

Comment:
why not adding the annotation with path to the networkType in cluster-crs.yaml-> AgentClusterInstall.
should be something like 
`  annotation:
    agent-install.openshift.io/install-config-overrides: siteconfig.Spec.Clusters.NetworkType
 `

Reply:
I couldn't do something like that in the ACI yaml, it seems not possible to add a spec value and a string in the same line, that's why this modifies the CR inside the getClusterCRs function.

Reply:
What I mean is ; We set the `siteConfigTemp.Spec.Clusters[clusterId].NetworkType = "{\"networking\":{\"networkType\":\"" + networkType + "\"}}" ` then as the annotation in ACI yaml have been defined as above it will just set the NetworkType value. 
AND please squash the commits into 1 commit and set the commit message start with **ztp:** the ci/proc/ci Job will continue to fail if u didn't

Reply:
That did not work, this is what I got in the generated file:

~~~yaml
kind: AgentClusterInstall
metadata:
    annotations:
        agent-install.openshift.io/install-config-overrides: siteConfigTemp.Spec.Clusters[clusterId].NetworkType = "{\"networking\":{\"networkType\":\"" + siteConfigTemp.Spec.Clusters[clusterId].NetworkType + "\"}}"

~~~

Reply:
The cluser-crs.yaml->AgentClusterInstall should as below
```
kind: AgentClusterInstall
  metadata:
    annotations:
      agent-install.openshift.io/install-config-overrides: siteConfig.Spec.Clusters.NetworkType 
```
 In the go implementation siteConfigBuilder.go Line56  as u did the changes instead of the checks and create the annotation map we can set the networkType as below
 
 `cluster.NetworkType = "{\"networking\":{\"networkType\":\"" + cluster.NetworkType + "\"}}"`
 
 this way when we append a new value to the annotation Map we will not need to re-create the annotation will just append to the cluster-crs.yaml and the crd definition.

Sorry for being not clear in the comment AND the CI job complain about go code style format. To avoid that in the repo root dir u can check the code format by run `$ make gofmt` to make sure that there is no missing space or style.

Reply:
Hey @serngawy,

It's not working, I believe mixing text (agent-install.openshift.io/install-config-overrides:) and var values (siteConfig.Spec.Clusters.NetworkType) in the same line on the template doesn't quite work.

This is what I get:

kind: AgentClusterInstall
metadata:
    annotations:
        agent-install.openshift.io/install-config-overrides: siteConfig.Spec.Clusters.NetworkType